lib/liquid_voting/delegations.ex

```diff
@@ -142,6 +142,48 @@ defmodule LiquidVoting.Delegations do
     |> Repo.insert()
   end
 
+  @doc """
+  Upserts a delegation (updates or inserts).
+
+  Updates existing global delegation for a specific delegator
+  if a new global delegation for the same delegator is passed in.
+  Updates existing proposal-specific delegation for a specific delegator
+  if a delegation for the same proposal and delegator is passed in.
+  Creates a new delegation if neither aforementioned condition is true.
+
+  ## Examples
+
+      iex> upsert_delegation(%{field: value})
+      {:ok, %Delegation{}}
+
+      iex> upsert_delegation(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+  """
+  def upsert_delegation(attrs \\ %{}) do
+    %{delegator_id: delegator_id} = attrs
+    proposal_url = get_or_set_proposal(attrs)
```

Comment:
I get that this syntax does the same thing, but why is it preferable?

Reply:
It's shorter (and from the std lib) and the logic is right there rather than in functions lower down. If you think this code is too opaque, you could make the default explicit as being `nil` in the third argument, though it is the default for the function and so probably unnecessary.

Reply:
Also, I didn't understand `get_or_set_proposal` before reading the code, I don't think a named function adds clarity.

Reply:
But my approach adds `proposal_url` key with value `nil` too `attrs` if not present in `attrs`. Would the approach you suggest do that too?

Reply:
actually, your approach works fine, and I can't remember why I thought I need the more complex approach.

Reply:
It's doing exactly the same thing 😉